### DISCUSSION TITLE
Required workflows only trigger on pull_request events

### DISCUSSION THREAD

**Discussion by fproulx-boostsecurity** [community/community#43889] [Actions]
### Select Topic Area

Product Feedback

### Body

Epic job on the new Required Workflows feature. This is nearly perfect, only if in addition to running on `pull_request` event , we could optionally have them run only on "merge to default branch". Running on the PR is good, but running whenever something gets actually merged to the default branch is very important to get scan results on the approved / merged version, getting a stable baseline scan.

**Comment by bobby-lin-cdc** [comment]
Agree. And also allow the required workflow to run on `push` and `schedule` events as well.

**Comment by ajmalab** [comment]
Can this not be solved by setting an `if` in your required workflow? Eg:

`if:  github.event.pull_request.merged == true`
